python/revng/scripting/projects/cli_project.py

```diff
@@ -0,0 +1,106 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import os
+import subprocess
+from tempfile import NamedTemporaryFile
+from typing import List
+
+import yaml
+
+from revng.model import Binary, DiffSet, Function  # type: ignore[attr-defined]
+
+from .project import Project
+
+
+class CLIProject(Project):
+    """
+    This class is used to run revng analysis and artifact through
+    the CLI with subprocess.
+    """
+
+    def __init__(self, resume_dir: str | None = None, revng_executable: str | None = None):
+        self._set_resume_dir(resume_dir)
+        self._set_revng_executable(revng_executable)
+        super().__init__()
+
+    def _load(self):
+        self._get_pipeline_description()
+        # TODO: We hardcode the path to the `model.yml`, fix when there is a
+        # command to get the model.
+        model_path = f"{self.resume_dir}/context/model.yml"
+        if not os.path.exists(model_path) and not os.path.isfile(model_path):
+            return
+        with open(model_path) as f:
+            self._set_model(Binary.deserialize(f))
+
+    def import_binary(self, input_binary_path: str):
+        self.input_binary = input_binary_path
+
+    def _get_artifact(self, artifact_name: str, targets_class: List[Function] = []) -> bytes | str:
+        targets = []
+        if targets_class:
+            kind = self._get_artifact_kind(artifact_name)
+            targets.extend([f"{tc.key()}:{kind}" for tc in targets_class])
+
+        result = self._get_artifact_internal(artifact_name, targets)
+        return self._get_result_mime(artifact_name, result)
+
+    def _get_artifact_internal(self, artifact_name: str, targets: List[str] = []) -> bytes | str:
+        args = [
+            self.revng_executable,
+            "artifact",
+            f"--resume={self.resume_dir}",
+            artifact_name,
+            self.input_binary,
+        ]
+        args.extend(targets)
+
+        return subprocess.run(args, capture_output=True, check=True).stdout
```

Comment:
Factor in a method code run `revng_executable`.
There, we should probably throw a nicer, custom exception in case of non-zero exit code.

Reply:
Done